src/libutil/terminal.cc

```diff
@@ -61,14 +61,28 @@ inline std::pair<int, size_t> charWidthUTF8Helper(std::string_view s)
 
 namespace nix {
 
-bool isTTY()
+bool isOutputARealTerminal(StandardOutputStream fileno)
 {
-    static const bool tty =
-        isatty(STDERR_FILENO)
-        && getEnv("TERM").value_or("dumb") != "dumb"
-        && !(getEnv("NO_COLOR").has_value() || getEnv("NOCOLOR").has_value());
+    return isatty(int(fileno)) && getEnv("TERM").value_or("dumb") != "dumb";
+}
 
-    return tty;
+bool shouldANSI(StandardOutputStream fileno)
+{
+    // Implements the behaviour described by https://bixense.com/clicolors/
+    // As well as https://force-color.org/ for compatibility, since it fits in the same shape.
+    // NO_COLOR CLICOLOR CLICOLOR_FORCE Colours?
+    // set      x        x              No
+    // unset    x        set            Yes
+    // unset    x        unset          If attached to a terminal
+    //                                  [we choose the "modern" approach of colour-by-default]
```

Comment:
I'm assuming the `x` means "any value"...? Could you just write that, please? :laughing: 

It took me so long to understand the difference between "x" and "set" x)